beacon/weakcoin/weak_coin.go

```diff
@@ -236,20 +240,20 @@ func (wc *WeakCoin) updateProposal(ctx context.Context, message Message) error {
 	return wc.updateSmallest(ctx, message.VrfSignature)
 }
 
-func (wc *WeakCoin) prepareProposal(epoch types.EpochID, nonce types.VRFPostIndex, round types.RoundID) ([]byte, []byte) {
+func (wc *WeakCoin) prepareProposal(epoch types.EpochID, nonce types.VRFPostIndex, round types.RoundID) ([]byte, types.VrfSignature) {
 	minerAllowance := wc.allowance.MinerAllowance(wc.epoch, wc.signer.NodeID())
 	if minerAllowance == 0 {
-		return nil, nil
+		return nil, types.EmptyVrfSignature
 	}
 	var broadcast []byte
-	var smallest []byte
+	smallest := new(big.Int).Lsh(big.NewInt(1), types.VrfSignatureSize*8)
 	for unit := uint32(0); unit < minerAllowance; unit++ {
 		proposal := wc.encodeProposal(epoch, nonce, round, unit)
 		signature := wc.signer.Sign(proposal)
 		if wc.aboveThreshold(signature) {
 			continue
 		}
-		if smallest == nil || bytes.Compare(signature, smallest) == -1 {
+		if new(big.Int).SetBytes(signature[:]).Cmp(smallest) == -1 {
```

Comment:
i think with above, we can do a simple bytes.Compare(signature[:], smallest)?

Reply:
Probably, I could get rid of `big.Int` and replace everything with `[]byte` or `[80]byte`, I just did it because I thought

```golang
smallest := new(big.Int).Lsh(big.NewInt(1), types.VrfSignatureSize*8)
```

is cleaner than

```golang
var smallest types.VrfSignature
for i := range smallest {
    smallest[i] = 0xff
}
```

but you are right the comparison would be simpler:

```golang
bytes.Compare(signature.Bytes(), smallest.Bytes()) == -1
```

vs

```golang
new(big.Int).SetBytes(signature[:]).Cmp(smallest) == -1
```

I'll switch back to arrays over `big.Int`